oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/PersistenceTimer.java

```diff
@@ -104,167 +92,51 @@ public void start(ModuleManager moduleManager, CoreModuleConfig moduleConfig) {
 
-        DefaultBlockingBatchQueue<PrepareRequest> prepareQueue = new DefaultBlockingBatchQueue(
-            this.maxSyncOperationNum);
-        try {
+        try (HistogramMetrics.Timer allTimer = allLatency.createTimer()) {
             List<PersistenceWorker<? extends StorageData>> persistenceWorkers = new ArrayList<>();
             persistenceWorkers.addAll(TopNStreamProcessor.getInstance().getPersistentWorkers());
             persistenceWorkers.addAll(MetricsStreamProcessor.getInstance().getPersistentWorkers());
 
-            // CountDownLatch makes sure all prepare threads done eventually.
-            CountDownLatch prepareStageCountDownLatch = new CountDownLatch(persistenceWorkers.size());
-
+            CountDownLatch countDownLatch = new CountDownLatch(persistenceWorkers.size());
             persistenceWorkers.forEach(worker -> {
                 prepareExecutorService.submit(() -> {
-                    if (stop.get()) {
-                        prepareStageCountDownLatch.countDown();
-                        return;
-                    }
-
-                    HistogramMetrics.Timer timer = prepareLatency.createTimer();
-                    try {
+                    try (HistogramMetrics.Timer timer = prepareLatency.createTimer()) {
                         if (log.isDebugEnabled()) {
                             log.debug("extract {} worker data and save", worker.getClass().getName());
                         }
                         List<PrepareRequest> innerPrepareRequests = new ArrayList<>(5000);
                         worker.buildBatchRequests(innerPrepareRequests);
-                        // Push the prepared requests into DefaultBlockingBatchQueue,
-                        // the executorService consumes from it when it reaches the size of batch.
-                        prepareQueue.offer(innerPrepareRequests);
-                        worker.endOfRound();
-                    } finally {
-                        timer.finish();
-                        prepareStageCountDownLatch.countDown();
-                    }
-                });
-            });
 
-            List<Future<?>> batchFutures = new ArrayList<>();
-            for (int i = 0; i < syncOperationThreadsNum; i++) {
-                Future<?> batchFuture = executorService.submit(() -> {
-                    // consume the metrics
-                    while (!stop.get()) {
-                        List<PrepareRequest> partition = prepareQueue.poll();
-                        if (partition.isEmpty()) {
-                            break;
-                        }
-                        HistogramMetrics.Timer executeLatencyTimer = executeLatency.createTimer();
-                        try {
-                            if (CollectionUtils.isNotEmpty(partition)) {
-                                batchDAO.flush(partition);
+                        try (HistogramMetrics.Timer executeLatencyTimer = executeLatency.createTimer()) {
```

Comment:
The meaning of `prepareLatency` has changed. It will contain `executeLatency`.

Reply:
I will fix this.

Reply:
@Ax1an I make the time of prepare stage accurate. And remove the outside preparedRequest list initialization, worker instance could always build a suitable size list in the first place (Reduce Array.copy and GC load a little).